src/components/Articles/ArticleFocus/ArticleFocus.jsx

```diff
@@ -0,0 +1,62 @@
+import { getSingleArticle, patchArticle } from '../../../utils/utils'
+import './ArticleFocus.css'
+import { useState, useEffect } from 'react'
+import { useParams, Link } from 'react-router-dom'
+
+const ArticleFocus = ({isLoading, setIsLoading}) => {
+    const [currArticle, setCurrArticle] = useState({})
+    const [hasVoted, setHasVoted] = useState(false)
+    const [localVotes, setLocalVotes] = useState(0)
+    const {article_id} = useParams()
+    
+    useEffect(() => {
+        setIsLoading(true)
+        getSingleArticle(article_id)
+        .then((article) => {
+            setCurrArticle(article)
+            setLocalVotes(article.votes)
+            setIsLoading(false)
+        })
+    }, [])
+    
+    const { title, author, body, created_at, comment_count, topic, article_img_url } = currArticle
+    const postedAt = new Date(created_at)
+
+    const handleVote = () => {
+        const vote = { inc_votes: !hasVoted ? 1 : -1}
+        setLocalVotes(localVotes + vote.inc_votes)
+        patchArticle(article_id, vote)
+        .then((article) => {
+            setLocalVotes(article.votes)
+            setHasVoted(!hasVoted)
+        })
+        .catch((err) => {
+            if (hasVoted) {
+                setLocalVotes(localVotes - 1)
+            } else {
+                setLocalVotes(localVotes + 1)
+            }
+            setHasVoted(!hasVoted)
+        })
```

Comment:
I am wondering why all this is in the catch block?

Reply:
My idea was that it's optimistically rendered until the results of the patch and if the patch fails it puts the button back to normal (voted/unvoted depending on what the user tried to do). I just broke the patch code and can see now that it's not correcting its previous optimism and the catch block isn't doing anything but I don't know why. I'm just really stuck on this.

Reply:
Hi Gilaine,  that was the right way to think about it -  add a catch block to the promise chain in your handle vote function, and inside it decrement the value of votes and set some state that will enable an error message to be shown to the user.  If the util function returned an error this catch block would be entered.  
I was just confused by the 'hasVoted' condition being dealt with in the catch as well.   
Your approach here is effectively doing the same thing, just relying on a particular return value from the util function if an error is returned.